File: interview-coach-frontend/src/types/resumeRewriter.ts
export interface BulletRewriteItem {
  original: string;
  improved: string;
  issue: string;
  ats_keywords_added: string[];
}

export interface ResumeRewriterResponse {
  overall_ats_score: number;
  ats_issues: string[];
  bullet_rewrites: BulletRewriteItem[];
  missing_metrics_summary: string;
  general_improvements: string[];
}

File: interview-coach-backend/schemas/resume_rewriter.py
from pydantic import BaseModel, Field


class BulletRewriteItem(BaseModel):
    original: str
    improved: str
    issue: str          # what was weak about it (e.g. "no metrics", "passive voice")
    ats_keywords_added: list[str] = Field(default_factory=list)


class ResumeRewriterResponse(BaseModel):
    overall_ats_score: int = Field(..., ge=0, le=100, description="ATS-friendliness score, 0-100")
    ats_issues: list[str] = Field(default_factory=list)
    bullet_rewrites: list[BulletRewriteItem] = Field(default_factory=list)
    missing_metrics_summary: str
    general_improvements: list[str] = Field(default_factory=list)